engine/src/test/java/org/kigalisim/engine/number/UnitConverterTest.java

```diff
@@ -0,0 +1,588 @@
+/**
+ * Unit tests for the UnitConverter class.
+ *
+ * @license BSD-3-Clause
+ */
+
+package org.kigalisim.engine.number;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.math.BigDecimal;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for the UnitConverter class.
+ */
+public class UnitConverterTest {
+
+  /**
+   * Mock implementation of StateGetter for testing.
+   */
+  private static class MockStateGetter implements StateGetter {
```

Comment:
Let’s use Mockito here but please use ignore validation so we can just set the values and verify correct behavior using the current assertions.